.github/workflows/on_push_ExtraJobsForMain.yml

```diff
@@ -43,4 +43,8 @@ jobs:
         run: |
           cd build
           ctest --output-on-failure
-          bash <(curl -s https://codecov.io/bash)
+          pip install gcovr
+          gcovr -r ./../ -x --exclude-unreachable-branches --exclude-throw-branches -o coverage.xml .
+          curl -Os https://uploader.codecov.io/latest/linux/codecov
+          chmod +x codecov
+          ./codecov -f build/coverage.xml
```

Comment:
Same as above

```suggestion
          curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --import
          curl -Os https://uploader.codecov.io/latest/linux/codecov
          curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
          curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig
          gpg --verify codecov.SHA256SUM.sig codecov.SHA256SUM
          shasum -a 256 -c codecov.SHA256SUM
          chmod +x codecov
          ./codecov -f build/coverage.xml
```

Reply:
@kevinbackhouse thanks having a look and catching that! :)